p2p/host/basic/basic_host.go

```diff
@@ -191,12 +194,54 @@ func NewHost(ctx context.Context, net network.Network, opts *HostOpts) (*BasicHo
 		h.pings = ping.NewPingService(h)
 	}
 
+	if h.introspector == nil {
+		h.introspector = introspection.NewDefaultIntrospector()
+	}
+
 	net.SetConnHandler(h.newConnHandler)
 	net.SetStreamHandler(h.newStreamHandler)
 
+	// register runtime provider
+	if err := h.introspector.RegisterProviders(&coreit.ProvidersTree{Runtime: h.runtimeDataProvider()}); err != nil {
+		log.Errorf("failed to register a runtime provider, err=%s", err)
+	}
+
+	// start introspection server
+	// TODO What happens if address is not configured or not available  ?
+	// TODO How do we configure a "default address"
```

Comment:
We need to cleanly manage the address on which the introspection server will run. It will need some thought.